azurelinuxagent/ga/exthandlers.py

```diff
@@ -1084,29 +1090,63 @@ def get_status_file_path(self, extension=None):
         return seq_no, path
 
     def collect_ext_status(self, ext):
-        self.logger.verbose("Collect extension status")
-
+        self.logger.verbose("Collect extension status for {0}".format(ext.name))
         seq_no, ext_status_file = self.get_status_file_path(ext)
         if seq_no == -1:
             return None
 
+        data = None
+        data_str = None
         ext_status = ExtensionStatus(seq_no=seq_no)
+
+        try:
+            data_str, data = self._read_and_parse_json_status_file(ext, seq_no, ext_status_file)
+        except ExtensionStatusError as e:
+            logger.periodic_warn(logger.EVERY_HALF_HOUR, u"[PERIODIC] " + ustr(e))
```

Comment:
we talked about periodic logging may hide errors when the incarnation changes and, *at the very least*, document this so that people don't mess up the messages unintentionally.

Reply:
As we discussed, the underlying messages raised by `_read_and_parse_json_status_file` for exception cases contain the information about the sequence number and other relevant information required in the logs.

But I understand your concern that the messages should be crafted when the logs are written, such that the messages are clear at the place they are logged. I'll bring the `msg` building up one level.
```
"We couldn't find any status for {0}-{1} extension, for the sequence number {2}. It failed due to {3}".format(ext.name, self.ext_handler.properties.version, seq_no, raised_exception)
```
```
msg = u"Failed to read any status for extension - {0}-{1}, Sequence number {2}. Failed due to {3}".format(ext.name, self.ext_handler.properties.version, seq_no, raised_exception)
```

Reply:
My concern is different. We can discuss offline.